pkg/cloud/gcp/gcs_storage.go

```diff
@@ -43,6 +45,9 @@ const (
 	// CredentialsParam is the query parameter for the base64-encoded contents of
 	// the Google Application Credentials JSON file.
 	CredentialsParam = "CREDENTIALS"
+	// TargetPrincipalParam is the target principal to impersonate. The target
```

Comment:
just for my edification what is a principal? Do you think it is worth spelling that out here?